lwt/nbd_lwt_unix.ml

```diff
@@ -54,7 +54,22 @@ let io_complete op fd buffer =
   then Lwt.fail End_of_file
   else return ()
 
-let tls_channel_of_fd fd role () =
+let impatient ?timeout_seconds f buf =
+  let timeout timeout_seconds =
+    Lwt_unix.sleep timeout_seconds >>= fun () ->
+    let msg = Printf.sprintf "Closing connection due to IO timeout: %f seconds" timeout_seconds in
+    Lwt_log.warning msg >>= fun () ->
+    Lwt.fail (Channel.Timeout timeout_seconds)
+  in
+  match timeout_seconds with
+  | None -> f buf
+  | Some interval -> (
+    if interval < 1.0e-9
+    then Lwt.fail_invalid_arg "Timeout must be at least one nanosecond; got %f s."
```

Comment:
Did you mean to use `sprintf` here? There is a stray '%f'

Reply:
Oops, yes.
I've force-pushed a fix now.